fecgov/fec-cms
# Remove legacy reporting dates JS

## Summary (required)
**NOTE:** Please merge this PR before merging https://github.com/fecgov/fec-cms/pull/6597
- Resolves #6491

- Remove the legacy `eporting-dates-tables.js`
- Rename the new `election-reporting-dates-tables.js` to `reporting-dates-tables.js` so template and JS files have the same name for clarity
- Remove references to JS in `full_width_page.html`

### Required reviewers

one frontend and **optionally** one content (@djgarr)

## Impacted areas of the application

Reporting dates tables from 2020 to present

deleted:    fec/static/js/pages/election-reporting-dates-tables.js
modified:   fec/static/js/pages/reporting-dates-tables.js
modified:   home/templates/home/full_width_page.html
modified:   home/templates/home/reporting_dates_table.html

### Related issues/PRs
Issues: 
https://github.com/fecgov/fec-cms/issues/5153

PRs:
New:
https://github.com/fecgov/fec-cms/pull/5786
https://github.com/fecgov/fec-cms/pull/5798
Legacy: 
https://github.com/fecgov/fec-cms/pull/3227

## How to test
- Restore your local db to the latest cms db dump
- checkout and run `npm run build-js`
- Test pages using `reporting_dates_table.html`  for JS functionality (footnotes, state dropdown, header notes)
- In wagtail admin:  
    - [Dates: 2020 reporting dates and deadlines](http://127.0.0.1:8000/admin/pages/10385/)
    - [Dates 2024 reporting dates and deadlines](http://127.0.0.1:8000/admin/pages/12876/)
- Example page:
    - http://127.0.0.1:8000/help-candidates-and-committees/dates-and-deadlines/2020-reporting-dates/congressional-pre-election-reporting-dates-2020/
    - http://127.0.0.1:8000/help-candidates-and-committees/dates-and-deadlines/dates-2024-reporting-dates-and-deadlines/congressional-pre-election-reporting-dates-2024/
- There were some legacy pages remaining in  full-width template  that needed inline JS to preserve zebra striping (see "Inline  JS sectio in the issue.  To confirm that zebra striping works on one of these pages, add the JS in an html block below the table's html block. 
     -  Example page on local:
    http://127.0.0.1:8000/ /help-candidates-and-committees/dates-and-deadlines/2020-reporting-dates/electioneering-communications-periods-special-elections-2020/
